Lesson3/Program.cs

```diff
@@ -1,12 +1,83 @@
-﻿using System;
+﻿using Project.Interface;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
 
-namespace Lesson3
+namespace Project.ConsoleApp
 {
-    class Program
+    public class Program
     {
-        static void Main(string[] args)
+        public static List<string> GetSortedData(string[] fileData)
         {
-            Console.WriteLine("New lesson 3 command line.");
+            List<string> sortedData = new List<string>();
+
+            //create a list then order it so nulls are last in the list
+            if (fileData != null)
+            {
+                sortedData = fileData
+                    .OrderBy(fd => fd == null)
+                    .ThenBy(fd => fd)
+                    .ToList();
+            }
+
+            return sortedData;
         }
+
+        [STAThread]
+        public static void Main(string[] clArguments)
+        {
+            if (clArguments.Any(a => String.Compare(a, "-json", true) == 0) || clArguments.Any(a => String.Compare(a, "-xml", true) == 0))
+            {
+                //use interface
+                IFileHandling ifhJson = GetJsonParser();
+                IFileHandling ifhXml = GetXmlParser();
+
+                List<string> parsedData = GetSortedData(ifhJson?.GetParsedData(GetFileData(GetFilePath("-json")))
+                                                .Concat(ifhXml?.GetParsedData(GetFileData(GetFilePath("-xml")))).ToArray())
+                                            .ToList();
+
+                parsedData?.ForEach(pd => Console.WriteLine("{0}", pd));
+            }
+            else
+            {
+                throw new ArgumentException("Invalid arguments.");
```

Comment:
`Observation`: `ArgumentException` can take a parameter to better explain which parameter is invalid.